### PR TITLE
 [Ikanji David]-Sage-Build a dapp for buying and selling unique gadgets on celo blockchain

### PR DESCRIPTION
**Pull request by Ikanji201**
### Thanks for submitting this tutorial for Celo Sage

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines) and reviewed the [Sample Tutorial Structure](https://docs.celo.org/community/celo-sage/tutorial-structure).

Please confirm the following before submitting your tutorial for review -

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the tutorial template structure.
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've run the tutorial to test it and the code works.
 
> Reviewers - @joenyzio & @viral-sangani

### CODE REVIEW

File: blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md [hunk 1]
```diff
@@ -0,0 +1,905 @@
...
+
+This code snippets defines a function called buyGadget, which allows a user to purchase a gadget from the system by passing in an index number and paying the price of the gadget in cUSD (a type of cryptocurrency).
+
+When someone calls the `buyGadget` function, it first creates a reference to the gadget being purchased using the Glass storage currentGlass = glasses[_index] statement.
+
+Next, the function checks to make sure that the person calling the function is not the owner of the gadget. If the condition is not met, the function stops executing and returns an error message saying "You can't buy your own gadget".
+
+Assuming the person calling the function is not the owner of the gadget, the function then attempts to transfer the price of the gadget in cUSD from the buyer to the owner of the gadget using the `IERC20Token(cUsdTokenAddress)`.transferFrom statement. If the transfer is successful, the function updates the owner of the gadget to be the buyer, and emits a buyGadgetEvent event to let other parts of the system know that a gadget has been purchased.
+
+Finally, for the smart contract we add the `Like` and the getGadgetLength` function.
```
Comment (4undRaiser) [inline blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md:429]: fix the formating for this line, there's no openning tag for getGadgetLength function

File: blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md [hunk 2]
```diff
@@ -0,0 +1,905 @@
...
+
+Lastly, we render the App component and return the Gadget components with the required props.
+
+## NEXT STEP
+
+I hope you found this tutorial informative and learned a lot from it. If you would like to further your education on this topic, here are some useful links that you can explore:
+
+[The official Celo documentation](https://docs.celo.org/)
+
+[Solidity By Example, a website with code examples for learning Solidity]( https://solidity-by-example.org/)
```
Comment (4undRaiser) [inline blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md:895]: remove the unnecessary space in the link

File: blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md [hunk 3]
```diff
@@ -0,0 +1,905 @@
...
+  - name: Ikanji David
+    title: Technical Writer 
+    url:   https://github.com/Ikanji201
+    image_url:  https://avatars.githubusercontent.com/u/115812158?v=4
+tags: [celo sage, dapp, intermediate, celo]
+hide_table_of_contents: true
+slug: /tutorials/build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain
+---
+
+![build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain](https://user-images.githubusercontent.com/115812158/229753542-ef563aa6-d625-4327-8137-512020e43f97.png)
```
Comment (4undRaiser) [inline blog/04-4-2023-Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain/Build-a-dapp-for-buying-and-selling-unique-gadgets-on-celo-blockchain.md:14]: Use this format for your header image instead.

header

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**